File: backend/internal/models/payment.go
package models

import "time"

type Payment struct {
	ID          int       `json:"id" gorm:"primary_key;AUTO_INCREMENT"`
	Uuid        string    `json:"uuid" gorm:"column:uuid;type:varchar(36)"`
	UserId      string    `json:"user_id" gorm:"column:user_id;type:varchar(36)"`
	Total       int       `json:"total" gorm:"column:total"`
	Type        string    `json:"type" gorm:"column:type"`
	Description string    `json:"description" gorm:"column:description"`
	SalesId     string    `json:"sales_id" gorm:"column:sales_id;type:varchar(36)"`
	PurchaseId  string    `json:"purchase_id" gorm:"column:purchase_id;type:varchar(36)"`
	Deleted     bool      `json:"deleted" gorm:"column:deleted"`
	CreatedAt   time.Time `json:"created_at" gorm:"column:created_at"`
	UpdatedAt   time.Time `json:"updated_at" gorm:"column:updated_at"`
}

func (*Payment) TableName() string {
	return "payment"
}

type PaymentResponse struct {
	Uuid        string    `json:"uuid"`
	UserId      string    `json:"user_id"`
	Total       int       `json:"total"`
	Type        string    `json:"type"`
	Description string    `json:"description"`
	SalesId     string    `json:"sales_id"`
	PurchaseId  string    `json:"purchase_id"`
	IsDeleted   bool      `json:"is_deleted"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type CashFlowResponse struct {
	Balance int               `json:"balance"`
	Payment []PaymentResponse `json:"payment"`
}

type CreateManualPaymentRequest struct {
	Total       int    `json:"total"`
	Type        string `json:"type"`
	Description string `json:"description"`
}

type CreatePaymentPurchaseRequest struct {
	PurchaseId   string    `json:"purchase_id" validate:"required"`
	PurchaseDate time.Time `json:"purchase_date" validate:"required"`
	StockCode    string    `json:"stock_code" validate:"required"`
	Total        int       `json:"total" validate:"required"`
}

type CreatePaymentSaleRequest struct {
	SalesId   string    `json:"sales_id" validate:"required"`
	SalesDate time.Time `json:"sales_date" validate:"required"`
	SalesCode string    `json:"sales_code" validate:"required"`
	Total     int       `json:"total" validate:"required"`
}

type UserBalanceDepositResponse struct {
	Balance int  `json:"balance"`
	Deposit bool `json:"deposit"`
}

File: frontend/src/types/payment.ts
export type PaymentStatus =
    | "ALL"
    | "PAYMENT_NOT_MADE_YET"
    | "PAYMENT_IN_FULL"
    | "PARTIAL_PAYMENT";

export const PaymentStatusLabel: Record<PaymentStatus, string> = {
    ALL: "Semua Status",
    PAYMENT_NOT_MADE_YET: "Belum Dibayar",
    PAYMENT_IN_FULL: "Lunas",
    PARTIAL_PAYMENT: "Sebagian",
};

export const MOCK_FILTER_STATUS_OPTIONS = [
    { key: "ALL", label: "Semua Status" },
    { key: "PAYMENT_IN_FULL", label: PaymentStatusLabel.PAYMENT_IN_FULL },
    { key: "PARTIAL_PAYMENT", label: PaymentStatusLabel.PARTIAL_PAYMENT },
    {
        key: "PAYMENT_NOT_MADE_YET",
        label: PaymentStatusLabel.PAYMENT_NOT_MADE_YET,
    },
];

export type PaymentType = "INCOME" | "EXPENSE";

export const PAYMENT_STATUS = {
    NOT_PAID: "PAYMENT_NOT_MADE_YET",
    FULL: "PAYMENT_IN_FULL",
    PARTIAL: "PARTIAL_PAYMENT",
} as const;

export interface CashFlowResponse {
    balance: number;
    payment: PaymentResponse[];
}

export interface PaymentResponse {
    uuid: string;
    user_id: string;
    total: number;
    type: PaymentType;
    description: string;
    sales_id: string;
    purchase_id: string;
    is_deleted: boolean;
    created_at: string;
    updated_at: string;
}

export interface ManualEntryFormRequest {
    tempId: string;
    total: number;
    type: PaymentType;
    description: string;
}

export interface CreatePaymentReqeust {
    purchase_id: string;
    purchase_date: string;
    stock_code: string;
    total: number;
}

export interface CreatePaymentSalesRequest {
    sales_id: string;
    sales_date: string;
    sales_code: string;
    total: number;
}

export interface UserBalanceDepositResponse {
    balance: number;
    deposit: boolean;
}
